mod_ood_proxy/lib/ood/nginx_stage.lua

```diff
@@ -1,15 +1,22 @@
+
 --[[
   pun
 
   Start PUN process for given user
 --]]
-function pun(r, bin, user, app_init_url)
+function pun(r, bin, user, app_init_url, exports, pre_hook_root_cmd)
   local cmd = bin .. " pun -u '" .. r:escape(user) .. "'"
+  local stdin = parse_exports(r, exports)
+
   if app_init_url then
     cmd = cmd .. " -a '" .. r:escape(app_init_url) .. "'"
   end
 
-  local err = capture2e(cmd)
```

Comment:
Can we take advantage of this STDIN any other way besides a prehook? Are we comfortable with changing every Execution of nginx_stage by Apache by providing STDIN like we are doing below.

If it could only ever be used for the pre_hook maybe we should consider maintaining the default `local err = capture2e(cmd)` and only passing stdin when a pre_hook_root_cmd is provided reduce the scope of this change. I have mixed feelings on this, though.